onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -0,0 +1,109 @@
+
+#include "core/optimizer/graph_transformer_utils.h"
+#include "core/optimizer/identity_elimination.h"
+#include "core/optimizer/slice_elimination.h"
+#include "core/optimizer/conv_mul_fusion.h"
+#include "core/optimizer/conv_bn_fusion.h"
+#include "core/optimizer/conv_add_fusion.h"
+#include "core/optimizer/unsqueeze_elimination.h"
+
+namespace onnxruntime {
+
+namespace transformerutils {
+
+Status ValidateTransformerLevel(unsigned int level) {
+  auto t_level = (unsigned int)std::pow(2, level);
+  if (t_level < TransformerLevel::MaxTransformerLevel) {
+    return Status::OK();
+  }
+  return Status(common::ONNXRUNTIME, common::FAIL, "Unsupported level " + level);
+}
+
+void SetTransformerContext(const uint32_t& level, uint32_t& levels_enabled, 
+                           std::vector<TransformerLevel>* all_levels) {
+
+  ORT_ENFORCE(ValidateTransformerLevel(level).IsOK(),
+              "Unsupported transformer level specified", level);
+
+  levels_enabled = TransformerLevel::Default;
+  if (level == 1) {
+    levels_enabled |= TransformerLevel::Level1;
+  } else if (level == 2) {
+    levels_enabled |= TransformerLevel::Level1 | TransformerLevel::Level2;
+  }
+
+  if (all_levels != nullptr) {
+    all_levels->push_back(TransformerLevel::Level2);
+    all_levels->push_back(TransformerLevel::Level1);
+  }
+}
```

Comment:
I think this should be a separate function so one method isn't trying to serve two different purposes (discover the valid levels and set the current level)